modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -1598,10 +1598,6 @@ def shuffle_partitions(
             ]
         ).T
         # We need to convert every partition that came from the splits into a full-axis column partition.
-        col_partitioning_func = np.vectorize(
-            lambda partition: cls._row_partition_class(partition)
-        )
-        split_row_partitions = col_partitioning_func(split_row_partitions)
         new_partitions = [
```

Comment:
Do we want to make this lazy? Since `split_row_partitions` is in effect the properly partitioned dataframe, we can transform to col partitions, and then `add_to_apply_calls` the sort instead, and defer metadata materialization till it's needed?

Reply:
This could be a future PR though!

Reply:
We can definitely do deferred meta-data materialization until really needed, created an issue for this #5808.

Regarding the lazy functions submission using `add_to_apply_calls`: I remember we had a performance regression quite ago when we switched to lazy execution. Since then we reverted the changes in #2471 and never tried to apply them again, so there is probably some careful evaluation that has to be done before making this change again, created an issue for this #5809